mbenfowler/sound-stash
# Feature/user conditions

This PR will:
- close #22 
- close #27 
- close #28 

## What I did:
- Implemented Context API (sorry, I forgot about my other branch and never opened up a PR for it. So it's included in this one)
- Adjusted, stubbed and expanded tests for Trending, Collections, Search and Login - sad paths included
- Refactored network fetch requests and implemented an `handleError` function to display the issue on the DOM
- Completed Demo user buttons to populate the login info
- Implemented functionality to allow non-logged-in users to view and search albums
- Implemented functionality to navigate non-logged-in users to the `/` path if the path is manually entered in the URL

## Testing Notes/Extras: 
- `npm run cypress` to test the Trending, Collections, Search and Login tests
- Sad network and non-existing pages tests are located in the Login tests
- Fixed Search tests to test for when there are no logged-in users
- Tested login features for all types of users and checked to see what would happen after logging in
- Tested feature page and what you would see with a logged-in user and no logged-in user
- tested Collection tests for multiple users
- All adjusted tests are stubbed. 

Did this break anything?


Type of change


Checklist:


What's next?
- Fix the styling on some elements. 
- Make sure all sad path tests are displayed on the dom
- Review and finalize testing
- Create GIFs to add to the readme
